sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
```

Comment:
Can't you check if it's an instance of ```MatrixBase``` and calculate the result normally? There weren't any problems with that before so I don't see why that wouldn't work here.